### ISSUE TITLE
(v1.3) exception in queued task error being caught.

### ISSUE THREAD

**Issue by vacarsu** [Urigo/angular-meteor#937]
```
Exception in queued task: TypeError: this.scope.$digest is not a function
    at ReactiveContext._propertyChanged (http://localhost:3000/packages/angular-meteor-data.js?824d3de5440cdbff4fbb313ba44222b234957eca:2387:22)
    at Object.cursor.observe.addedAt (http://localhost:3000/packages/angular-meteor-data.js?824d3de5440cdbff4fbb313ba44222b234957eca:2260:21)
    at observeChangesCallbacks.addedBefore (http://localhost:3000/packages/minimongo.js?cdf1a26cf7719fa9471a8017c3defd5aea812727:3623:28)
    at Object.self.applyChange.addedBefore (http://localhost:3000/packages/minimongo.js?cdf1a26cf7719fa9471a8017c3defd5aea812727:3560:56)
    at http://localhost:3000/packages/minimongo.js?cdf1a26cf7719fa9471a8017c3defd5aea812727:407:13
    at _.extend.runTask (http://localhost:3000/packages/meteor.js?9730f4ff059088b3f7f14c0672d155218a1802d4:685:11)
    at _.extend.flush (http://localhost:3000/packages/meteor.js?9730f4ff059088b3f7f14c0672d155218a1802d4:713:10)
    at _.extend.drain (http://localhost:3000/packages/meteor.js?9730f4ff059088b3f7f14c0672d155218a1802d4:721:12)
    at LocalCollection.insert (http://localhost:3000/packages/minimongo.js?cdf1a26cf7719fa9471a8017c3defd5aea812727:615:22)
    at Object.self._connection.registerStore.update (http://localhost:3000/packages/mongo.js?69942a86515ec397dfd8cbb0a151a0eefdd9560d:250:28)
```
It happens when the component loads and when I change the sort variable. When the sort variable changes the subscription doesn't reload data.
``` javascript
this.sort = { 'profile.firstName': 1 };
this.helpers({
  users() { return Meteor.users.find({}, { sort: this.sort }) },
  sort: this.sort
});
 this.subscribe('users');
```

**Comment by MilosStanic** [comment]
did you do `$reactive(this).attach($scope)` ?

**Comment by vacarsu** [comment]
I'm using angular2-now and extending the reactive component class. Perhaps there is something wrong with that package.

**Comment by magbeat** [comment]
I also get this error. I tracked it down to angular-meteor-data.js file and line 2387. The ```propertyChanged``` function gets called each time a new document is added to the collection which is used as a property in the controller.

```
vm.helpers({
   mortgages: function() { return Mortgages.find() }
});
```

and then later on:

```
vm.subscribe('mortgages', function() { return [] });
```

I am using the controllerAs convention.

**Comment by MilosStanic** [comment]
@vacarsu 
why don't you do like this: 
```javascript
this.helpers({
  users() { return Meteor.users.find({}, { sort: this.sort }) },
  sort: { 'profile.firstName': 1 }
});
 this.subscribe('users');
```

In your code you have `this.sort` declared, and then in `this.helpers` you declare it again. It's selfreferencing.

**Comment by vacarsu** [comment]
It has nothing to do with that. It has to do with the $digest not being applied. It resubscribe but the view does not get updated properly.

**Comment by magbeat** [comment]
@vacarsu could you try this:

```
this.sort = undefined;
this.helpers({
  users() { return Meteor.users.find({}, { sort: this.sort }) },
  sort:  { 'profile.firstName': 1 };
});
 this.subscribe('users');
```

My coworker had the same problem, when he 'initialized' the var outside the helper block.

**Comment by vacarsu** [comment]
@magbeat no dice. Still not working properly.

**Comment by dotansimha** [comment]
Can you provide a git for your project? 
Thanks

**Comment by MilosStanic** [comment]
@vacarsu is right.
I just started facing the same issue when I switched one of my components/controllers from v1.2 to v1.3. 
There are 25 documents in a collection I'm subscribing to, and when each doc is added to subscription, this error is fired.

**Comment by vacarsu** [comment]
@dotansimha I don't have time to create a reproduction at this time. I reverted the component I was testing on back to the old syntax for now. I was also having issues with components not even loading after notification. No errors being thrown in that case.

**Comment by MilosStanic** [comment]
this seems like the same error as in #943

**Comment by Urigo** [comment]
Fixed in the new [1.3.1 release](https://github.com/Urigo/angular-meteor/releases)
